saleor/checkout/utils.py

```diff
@@ -258,25 +271,23 @@ def _append_line_to_update(
             to_update.append(line)
 
 
-def _append_line_to_delete(to_delete, variant, line_data, variant_ids_in_lines):
-    if variant.pk not in variant_ids_in_lines:
+def _append_line_to_delete(to_delete, line_data, line):
+    if line is None:
         return
-    line = variant_ids_in_lines[variant.pk]
+
     quantity = line_data.quantity
     if line_data.quantity_to_update:
         if quantity <= 0:
             to_delete.append(line)
 
 
-def _append_line_to_create(
-    to_create, checkout, variant, line_data, variant_ids_in_lines
-):
-    if variant.pk not in variant_ids_in_lines:
+def _append_line_to_create(to_create, checkout, variants_map, line_data, line):
+    if line is None:
```

Comment:
It looks like you added `if line is None` to all helpers functions called in `add_variants_to_checkout(`
Wouldn't be good to move `if line is None` to the `for`:
```
    for line_data in checkout_lines_data:
         line = _get_line_if_exist(line_data, lines_by_id)
         if line is None:
             continue
```